onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -226,7 +226,7 @@ Status PadCpuImpl(OpKernelContext* ctx,
   input_starts.reserve(2 * new_dims_count);
   input_extents.reserve(2 * new_dims_count);
   for (size_t i = 0; i < new_dims_count; i++) {
-    input_starts.push_back(reshaped_slice[i]);
+    input_starts.push_back(-1 * reshaped_slice[i]);
```

Comment:
`reshaped_slice` holds negative values (by definition), so the start offset should move in the positive direction to skip over the input values that were "padded out"